ember/app/ui/components/identity-members/component.js

```diff
@@ -0,0 +1,46 @@
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+import { restartableTask, lastValue } from "ember-concurrency-decorators";
+
+/**
+ * @arg identity
+ */
+export default class IdentityMembersComponent extends Component {
+  @service store;
+  @service intl;
+  @service notification;
+
+  @lastValue("fetchMembers") members;
+  @restartableTask *fetchMembers() {
+    const members = yield this.store.query("membership", {
+      filter: { organisation: this.args.identity.id },
+      include: "role",
+    });
+
+    const membersDeduped = [];
+
+    members.forEach((member) => {
+      const existingMember = membersDeduped.findBy(
+        "identity.id",
+        member.identity.get("id")
+      );
+
+      if (existingMember && member.role.get("title")) {
+        existingMember.roles = [
+          ...existingMember.roles,
+          { title: member.role.get("title"), inactive: member.isInactive },
+        ];
+      } else {
+        member.roles = [
+          { title: member.role.get("title"), inactive: member.isInactive },
+        ];
+      }
```

Comment:
Maybe simpler:
```js
     member.roles = [
          { title: member.role.get("title"), inactive: member.isInactive },
      ];
      if (existingMember) {
        existingMember.roles = [
          ...existingMember.roles,
          ...member.roles,
        ];
      }
```
I omitted the check if the member's role has a title, because I was confused that the check is needed for existing members but not in the `else` case.

Reply:
The check for title is to prevent empty entries appearing, but the rest works and definitely looks better.